pkg/pipeline/write/write_loki.go

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package write
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/netobserv/flowlogs2metrics/pkg/api"
+	pUtils "github.com/netobserv/flowlogs2metrics/pkg/pipeline/utils"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/netobserv/flowlogs2metrics/pkg/config"
+
+	logAdapter "github.com/go-kit/kit/log/logrus"
+	jsonIter "github.com/json-iterator/go"
+	"github.com/netobserv/loki-client-go/loki"
+	"github.com/netobserv/loki-client-go/pkg/backoff"
+	"github.com/netobserv/loki-client-go/pkg/urlutil"
+	"github.com/prometheus/common/model"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	keyReplacer = strings.NewReplacer("/", "_", ".", "_", "-", "_")
+)
+
+type emitter interface {
+	Handle(labels model.LabelSet, timestamp time.Time, record string) error
+}
+
+const channelSize = 1000
+
+// Loki record writer
+type Loki struct {
+	lokiConfig loki.Config
+	apiConfig  api.WriteLoki
+	client     emitter
+	timeNow    func() time.Time
+	ready      bool
```

Comment:
In a later PR I suggested to remove the "ready" field: https://github.com/netobserv/goflow2-kube-enricher/pull/29/files#diff-f0ee701b73c37f99481d73281b339c6abe6d17c2d5a1f494f80edd16dacb7e79L59-L72

the reason is that this field does not provide any changing state and it is only used to override the behavior with the "empty loki" collector, which is only used for testing. I think we shouldn't permeate testing behavior into production code.

We can just remove this `ready` field and `IsReady` method.